app/controllers/hyrax/depositors_controller.rb

```diff
@@ -15,14 +14,18 @@ def index
     end
 
     def create
-      grantor = authorize_and_return_grantor
-      grantee = ::User.from_url_component(params[:grantee_id])
-      if grantor.can_receive_deposits_from.include?(grantee)
-        head :ok
+      if (params[:user_id].sub '.', '-dot-') == (params[:grantee_id].sub '.', '-dot-')
+        render json: { name: "FAILURE", status: :bad_request }
```

Comment:
How about returning the error message in the json and making the status be the http response code instead of inside the json?  It looks like Hyrax already provides a controller method to do this.  Also I think 422 is probably a better fit here than 400.
```suggestion
        return render_json_response(response_type: :unprocessable_entity, message: view_context.t('hyrax.dashboard.proxy_add_deny'))
```
Your JS would have to handle this differently since it wouldn't be a success(200) but I think this is more in line with what Hyrax is already doing and allows your modal to be more generic since it can use whatever the json response message was.